src/components/ui/Panel.jsx

```diff
@@ -0,0 +1,151 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+function getClosestIndex(arr, goal) {
+  return arr.reduce((closestIndex, curr, index) =>
+    Math.abs(curr - goal) < Math.abs(arr[closestIndex] - goal) ? index : closestIndex
+  );
+}
+
+export default class Panel extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    title: PropTypes.node,
+    minimizedTitle: PropTypes.node,
```

Comment:
c'est quoi node?

Reply:
Sorry, I should have explained the PropTypes stuff.
Here is the doc: https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes
`node` is basically everything that can be rendered. It's useful in this kind of context to be able to pass indifferently a simple string or a complex HTML markup